### ISSUE TITLE
Jupyter notebooks to another repository?

### ISSUE THREAD

**Issue by cmgrote** [odpi/egeria#5464]
At the moment our Jupyter notebooks for the various labs are all embedded in this core repository -- as with other examples (Helm charts, samples, etc) this means that we take a very long hit before we can merge anything while we wait for Java code to build (which the notebooks don't touch at all)...

I vaguely recall we'd previously discussing moving these out to e.g. the `egeria-samples` repository (or perhaps some other repository), but that we were waiting until some of the other dependent material was moved (like the Helm charts) (?)

Now that the Helm charts are in their own dedicated repository and no longer embedded here, would now be a logical time to move these lab notebooks out as well -- or was there something else we were dependent on / waiting on (?)

**Comment by planetf1** [comment]
Yes we have discussed, but indeed it was partly impacted by where the charts where, though that was never a hard block since the charts just depend on a docker image, more figuring out the logical split.  We also still have the 'docker-compose' as part of core egeria (which uses the notebook charts). I'd be tempted to deprecate them, leave as is, and no develop further, but that needs discussion. An alternative is to move them outside, perhaps even their own repo.

So I don't see a notable impediment to moving the charts. This should include the creation of the jupyter docker image (or an alternative approach that delivers the right content), the main question would be where to
 egeria-samples - have discussed for placing a variety of samples
 a new dedicated repo more targetted, just for coco
 a new dedicated repo specifically for charts

One thing to note is that new developments in the base are sometimes picked up in these notebooks quickly, so we have to be careful in terms of rebuilding etc.

Finally I won't be able to review or work on this during the remainder of the 3.0 release cycle, but am happy to talk through what is needed, testing requirements, prep for release pipelines etc if needed in the next few days.

**Comment by cmgrote** [comment]
Challenges with moving at the moment:

- Coco Pharma security setup is in the code itself at the moment (e.g. `CocoPharmaServerSecurityConnector`, etc)

**Comment by planetf1** [comment]
The notebooks could still be moved out for packaging, even if the security connector is in the base. It's arguably logically incomplete since the coco pharma enviroment is to be seen as a 'whole' but there's certainly no blocking issue to the move of the notebooks if we feel that is important and needed. - as per previous comment.
